src/Trajectory.cpp

```diff
@@ -27,16 +29,65 @@ using namespace chemfiles;
 
 #define SENTINEL_VALUE (static_cast<size_t>(-1))
 
+std::string chemfiles::guess_format(std::string path) {
+    std::string extension;
+    std::string compression;
+
+    auto dot1 = path.rfind('.');
+    if (dot1 != std::string::npos) {
+        extension = path.substr(dot1);
+        bool new_extension = false;
+        // check file extension for compressed file extension
+        if (extension == ".gz") {
+            new_extension = true;
+            compression = "GZ";
+        } else if (extension == ".bz2") {
+            new_extension = true;
+            compression = "BZ2";
+        } else if (extension == ".xz") {
+            new_extension = true;
+            compression = "XZ";
+        }
+
+        if (new_extension) {
+            extension = "";
+            auto dot2 = path.substr(0, dot1).rfind('.');
+            if (dot2 != std::string::npos) {
+                extension = path.substr(0, dot1).substr(dot2);
+            }
+        }
+    }
+
+    if (extension.empty()) {
+        throw file_error(
+            "file at '{}' does not have an extension, provide a format name to read it",
+            path
+        );
+    }
+
+    auto registered_format = FormatFactory::get().by_extension(extension);
+    auto format = std::string(registered_format.metadata.name);
+
+    if (!compression.empty()) {
+        format += " / " + compression;
+    }
```

Comment:
AFAIK there is no way to check whether or not the format actually supports the compression. Previously, this wouldn't matter because trying to open something like "traj.xtc.gz" would [thrown an error in the constructor](https://github.com/chemfiles/chemfiles/blob/c87c3dca55888540430031ab6a3297cea1784e9c/src/formats/XTC.cpp#L58-L60) immediately. But `chemfiles::guess_format("traj.xtc.gz");` happily returns "XTC / GZ", which introduces the subtle distinction between formats that are recognized (but not necessarily supported) and actually supported formats. However, it might be only me who would misuse such a function to check if a certain file is supported :)
Would it be helpful to highlight this fact in the docs?

Reply:
At the very least we should indeed explicitly document this behavior! 

My use case for this function mean that I will immediately call the trajectory constructor after using it, but I can see how it would be useful to check if a file would be supported. However, the fact that the extension of a file might be supported does not mean that the file is in the right format. For example, `.nc` files will be interpreted as following the Amber NetCDF convention, but such file could be any NetCDF file, meaning trying to open the file will fail. So for now I think keeping (and documenting) the current behavior (`guess_format` might return a format that later fails to open if the file is not formatted as expected, or uses unsupported compression scheme) should be fine.

Reply:
Updated!